code/modules/cargo/exports.dm

```diff
@@ -183,6 +201,46 @@ Credit dupes that require a lot of manual work shouldn't be removed, unless they
 
 GLOBAL_LIST_EMPTY(exports_list)
 
+/datum/export/proc/get_material_cost(atom/movable/A)
+	var/value = 0
+	if(A.custom_materials)
+		for(var/i in A.custom_materials)
+			var/datum/material/M = i
+			value += M.value_per_unit * A.custom_materials[M]
+	value = round(value)
+	return value
+
+/datum/export/proc/sell_material_item(obj/O, datum/export_report/report, dry_run = TRUE, apply_elastic = TRUE)
+	///This is the value of the object, as derived from material datums.
+	var/material_cost = get_material_cost(O)
+	///Quantity of the object in question.
+	var/amount = get_amount(O)
+	///Utilized in the pricetag component. Splits the object's profit when it has a pricetag by the specified amount.
+	var/profit_ratio = 0
+
+	if(amount <=0 || material_cost <=0)
+		return FALSE
+	unit_name = "[O.name]"
+	if(dry_run == FALSE)
+		if(SEND_SIGNAL(O, COMSIG_ITEM_SOLD, item_value = material_cost & COMSIG_ITEM_SPLIT_VALUE))
+			profit_ratio = SEND_SIGNAL(O, COMSIG_ITEM_SPLIT_PROFIT)
+			material_cost = material_cost*((100-profit_ratio)/100)
```

Comment:
```suggestion
			material_cost = material_cost * ((100 - profit_ratio) * 0.01)
```
Prefer the use of multiplications over divisions.
There's no need to save up on spaces, by the byes. Clarity is nice.